spring-integration-jms/src/main/java/org/springframework/integration/jms/JmsOutboundGateway.java

```diff
@@ -144,6 +145,12 @@
 
 	private volatile boolean requiresReply;
 
+	private long lastSend;
+
+	private volatile long idleReplyContainerTimeout;
```

Comment:
I'd say that `volatile` must be present exactly on the other variable rather than those which is populated through setters. It's rare operation, but not those which is done from the `handleRequestMessage` and that scheduled task.
Or don't I just understand the `volatile` premise?..